kikuchipy/signals/ebsd.py

```diff
@@ -1193,32 +1302,46 @@ def average_neighbour_patterns(
         # Create dask array of signal patterns and do processing on this
         dask_array = get_dask_array(signal=self)
 
+        # print(dask_array.chunksize)
+        # dask_array = dask_array.rechunk({0: "auto", 1: "auto", 2: -1, 3: -1})
+
         # Add signal dimensions to array be able to use with Dask's map_blocks()
         nav_dim = self.axes_manager.navigation_dimension
         for i in range(sig_dim):
             window_sums = np.expand_dims(window_sums, axis=window_sums.ndim)
         window_sums = da.from_array(
             window_sums, chunks=dask_array.chunksize[:nav_dim] + (1,) * sig_dim
```

Comment:
The reason for the error in #230 was this line, it should be:

```python
        window_sums = da.from_array(
            window_sums, chunks=dask_array.chunks[:nav_dim] + (1,) * sig_dim
        )
```

I'll push this change now.

Reply:
I've added a test with the exact data shape and chunk sizes as a signal where I saw a *similar* error to the one in #230. The error message in that test before the fix was a ValueError, not IndexError. but after making the fix the test passed. Also, after the fix, I didn't encounter the IndexError where I previously did. I'm therefore confident that this fix... fixed it.